akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentBehaviorSpec.scala

```diff
@@ -403,12 +413,27 @@ class PersistentBehaviorSpec extends ActorTestKit with TypedAkkaSpecWithShutdown
       watchProbe.expectMessage("Terminated")
 
       val probeC2 = TestProbe[(State, Event)]()
-      val c2 = spawn(counterWithProbe("c12", probeC2.ref).snapshotEvery(2))
+      val c2 = spawn(counterWithProbe(pid, probeC2.ref).snapshotEvery(2))
       probeC2.expectNoMessage()
       c2 ! GetValue(replyProbe.ref)
       replyProbe.expectMessage(State(3, Vector(0, 1, 2)))
     }
 
+    "wrap persistent behavior in tap" in {
+      val probe = TestProbe[String]
+      val wrapped: Behavior[Command] = Behaviors.tap(counter(nextPid))(
+        (_, _) => probe.ref ! "msg received",
```

Comment:
where do the journal messages (not instanceOf[Command]) end up? They shouldn't be received here?

was this a reproducer?

Reply:
Yes this was a reproducer. 

`tap` uses `Intercept` which uses a class tag to only pass in things that match the U type param. However tap used to use Any as the classtag rather than require one so every message was intercepted and passed to this function. Now only ones that match the Command are.

Reply:
ah, there it is `case m: U ⇒`